### PR TITLE
Semi-Automatic Construction

### PR DESCRIPTION
**Pull request by Spedwards**
Pray for no issues.

### CODE REVIEW

File: global.js [hunk 1]
```diff
@@ -64,6 +64,46 @@ mod.FLAG_COLOR = {
...
+        },
+        nuker: { // construct a nuker at flag when available
+            color: COLOR_CYAN,
+            secondaryColor: COLOR_GREY,
+        },
+        powerSpawn: { // construct a power spawn at flagwhen available
+            color: COLOR_CYAN,
+            secondaryColor: COLOR_WHITE,
+        }
+    },
```
Comment (karlthepagan) [inline global.js]: These are all reasonably good choices.

**Comment by Spedwards** [comment]
Currently unable to test this myself as I'm nowhere near a new RCL, however this should work.

There is a possibility that I have not thought of a case where a construction site cannot be created, in which case the `createConstructionSite` call will fail, but the flag will be removed. I will resolve this in the next commit.

**Comment by jisaf** [comment]
I'm about to launch 2 new rooms, anything specific you want tested?

**Comment by Spedwards** [comment]
@jisaf The flags in general. If one lot works, they all should, but I would just like to know.

Also theoretically the structures should get created in the order the flags are placed. That's an important thing I want to know. So if you could place 10-20 extension flags and take note of the order they're created.

Should be pretty straight forward.

**Comment by jisaf** [comment]
https://screeps.com/a/#!/room/E13N85

Should hit level 3 in a couple hours. Will mirror the extensions that are already there and add a tower at 37,34

I added a few extra extension flags randomly around, just for fun.

I'm not currently tracking extra CPU usage.

**Comment by jisaf** [comment]
So, I'm pretty sure I've passed 500 ticks since leveling up, with no new construction sites.

I'm going to leave it over night and see what happens, but I didn't change any defaults (both parameters set to true, using cyan/cyan for the extensions and cyan/purple for the tower)

**Comment by jisaf** [comment]
Error in room.js (Room.prototype.loop) for "E13N85": 
TypeError: Cannot read property 'lookFor' of undefined
    at CONSTRUCT (room:1586:35)
    at FlagDir.filter.splice.forEach.flag (room:1596:17)
    at Array.forEach (native)
    at Room.processConstructionFlags (room:1595:82)
    at getEnvironment (room:2204:22)
    at /opt/engine/node_modules/lodash/index.js:3073:15
    at baseForOwn (/opt/engine/node_modules/lodash/index.js:2046:14)
    at /opt/engine/node_modules/lodash/index.js:3043:18
    at Function. (/opt/engine/node_modules/lodash/index.js:3346:13)
    at Function.mod.analyze (room:2219:7)
TypeError: Cannot read property 'lookFor' of undefined
    at CONSTRUCT (room:1586:35)
    at FlagDir.filter.splice.forEach.flag (room:1596:17)
    at Array.forEach (native)
    at Room.processConstructionFlags (room:1595:82)
    at getEnvironment (room:2204:22)
    at /opt/engine/node_modules/lodash/index.js:3073:15
    at baseForOwn (/opt/engine/node_modules/lodash/index.js:2046:14)
    at /opt/engine/node_modules/lodash/index.js:3043:18
    at Function. (/opt/engine/node_modules/lodash/index.js:3346:13)
    at Function.mod.analyze (room:2219:7)

**Comment by chobobobo** [comment]
I'm using this to build all buildings in my new room, extensions work

**Pull request merged**